SpectralCubeReprojectMaskExample_v3.ipynb

```diff
@@ -0,0 +1,1546 @@
+      "text/plain": [
+       "<Figure size 432x288 with 1 Axes>"
+      ]
+     },
+     "metadata": {
+      "needs_background": "light"
+     },
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "cube2.max(axis=0).quicklook()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Step 4. Spectral Interpolation\n",
+    "\n",
+    "We can do the spatial or spectral step first.  In this case, we choose spectral.\n",
+    "\n",
+    "We need to match resolution:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 15,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "(<Quantity 2.00262828 km / s>, <Quantity 0.67429189 km / s>)"
+      ]
+     },
+     "execution_count": 15,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "velocity_res_1 = np.diff(cube1vel.spectral_axis)[0]\n",
+    "velocity_res_2 = np.diff(cube2vel.spectral_axis)[0]\n",
+    "velocity_res_1, velocity_res_2"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Cube1, our H2CS cube, has broader channels.  We want to smooth cube2 to that channel width before doing the spatial reprojection.\n",
+    "\n",
+    "To do this, we'll smooth with a Gaussian with an appropriate FWHM:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 16,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/latex": [
+       "$1.8856963 \\; \\mathrm{\\frac{km}{s}}$"
+      ],
+      "text/plain": [
+       "<Quantity 1.88569628 km / s>"
+      ]
+     },
+     "execution_count": 16,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "fwhm_gaussian = (velocity_res_1**2 - velocity_res_2**2)**0.5\n",
+    "fwhm_gaussian"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 17,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from astropy.convolution import Gaussian1DKernel\n",
+    "fwhm_to_sigma = np.sqrt(8*np.log(2))\n",
+    "# we want the kernel in pixel units, so we force to km/s and take the value\n",
+    "spectral_smoothing_kernel = Gaussian1DKernel(stddev=fwhm_gaussian.to(u.km/u.s).value / fwhm_to_sigma)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Then, let's cut down to the same spectral coverage:"
```

Comment:
You could start with this part as it intersects the cells on smoothing.

A brief explanation that this is not technically necessary, but is more computationally efficient to avoid smoothing/interpolation over a wider spectral range.

I think it is also worth expanding the cutout part to be more general, so handling the potential case where either `cube1` or `cube2` would need to be cutout to find the overlap. This could be pretty simple with just printing out the `cube1.spectral_extrema`, `cube2.spectral_extrema` to manually check for which cube(s) would need to be cutout.

Reply:
I have edited the code to be entirely general: now, both cubes get cut to the smallest limiting velocity range (even though one cut is redundant). I hope this is okay—it's a dumb way, but hopefully shouldn't take more computational time than checking which cube would need to be cut. Also, this takes care of the case where both cubes would need to be cut: if one is not entirely contained by the other.